### PR TITLE
Add evm error

### PR DESCRIPTION
**Pull request by rem1niscence**
Adds evm error and refactors a bit for when the need to add non evm error arise

### CODE REVIEW

File: src/utils/errors.ts [hunk 1]
```diff
@@ -3,14 +3,20 @@ import { parseJSONRPCError } from './parsing'
 export const EVM_ERROR_CODES = [
   // JSON RPC Standard errors
   '-32',
+  // Execution reverted error
+  '-3',
```
Comment (nymd) [inline src/utils/errors.ts]: doesn't this automatically include the line above?

**Review by crisog — APPROVED** [review]
LGTM

File: src/utils/errors.ts (hunk 1 above)
Comment (rem1niscence) [reply src/utils/errors.ts]: no as it checks for exact matches

Comment (crisog) [reply src/utils/errors.ts]: Yes, kind of a coincidence though. There are also custom codes 1 and wouldn't match -32.

Comment (crisog) [reply src/utils/errors.ts]: > no as it checks for exact matches

Why not? It's `.includes()`

Comment (crisog) [reply src/utils/errors.ts]: The error should be 3 and not -3 btw

Comment (rem1niscence) [reply src/utils/errors.ts]: yep, my bad I got confused, on that case is not working as expected, back to draft

Comment (rem1niscence) [reply src/utils/errors.ts]: > The error should be 3 and not -3 btw

oh that's why yes

File: src/utils/errors.ts [hunk 2]
```diff
@@ -3,14 +3,20 @@ import { parseJSONRPCError } from './parsing'
...
 export function isEVMError(response: string): boolean {
+  return isError(response, EVM_ERROR_CODES)
+}
+
+function isError(response: string, errorsToCheck: string[]): boolean {
   try {
     const { code } = parseJSONRPCError(response)
 
-    return EVM_ERROR_CODES.some((error) => String(code).includes(error))
+    return errorsToCheck.some((error) => String(code).includes(error))
```
Comment (crisog) [inline src/utils/errors.ts:19]: I think we should remove the string cast here, so we can have exact matches

**Pull request merged**